client/client-lib/src/api.rs

```diff
@@ -100,10 +100,10 @@ pub trait IFederationApi: Debug + Send + Sync {
 }
 
 dyn_newtype_define! {
-    pub FederationApi(Arc<IFederationApi>)
+    pub DynIFederationApi(Arc<IFederationApi>)
```

Comment:
I thought we are going to do `DynFederationApi` (here and elsewhere). The `I` seems needless. I'm already a bit sad about that `Dyn`. ;)

Reply:
oh okay  :) well, at least it's not a very long prefix